llvm/lib/Target/AArch64/GISel/AArch64RegisterBankInfo.cpp

```diff
@@ -605,6 +605,35 @@ bool AArch64RegisterBankInfo::onlyDefinesFP(const MachineInstr &MI,
   return hasFPConstraints(MI, MRI, TRI, Depth);
 }
 
+bool AArch64RegisterBankInfo::isLoadFromFPType(const MachineInstr &MI) const {
+  // GMemOperation because we also want to match indexed loads.
+  auto *Load = dyn_cast<GMemOperation>(&MI);
+
+  const auto &MMO = Load->getMMO();
```

Comment:
`dyn_cast` followed by unconditional dereference. Either check the result & return, or change to a `cast<>`